### PR TITLE
Update plugin test matrix GPU workflow names

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** Update plugin test matrix GPU workflow names.
Links:
[latest/latest](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/10960947889)
[latest/stable](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/10960949857)
[stable/latest](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/10960951835)
[stable/stable](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/10963394921) 

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,9 @@
 
 ### Improvements
 
+* Update plugin test matrix GPU workflow names.
+  [(#918)](https://github.com/PennyLaneAI/pennylane-lightning/pull/918)
+
```
Comment (maliasadi) [inline .github/CHANGELOG.md:39]: This has to be technically fixed in https://github.com/PennyLaneAI/pennylane-lightning/pull/911! We don't need two entrypoints in the changelog for a single task.

File: .github/workflows/compat-check-release-release.yml [hunk 2]
```diff
@@ -48,4 +48,3 @@ jobs:
       lightning-version: release
       pennylane-version: release
     secrets: inherit
```
Comment (maliasadi) [inline .github/workflows/compat-check-release-release.yml:50]: To fix the EOF rendering issue
```suggestion
    secrets: inherit

```

File: .github/workflows/compat-check-stable-stable.yml [hunk 3]
```diff
@@ -48,4 +48,3 @@ jobs:
       lightning-version: stable
       pennylane-version: stable
     secrets: inherit
```
Comment (maliasadi) [inline .github/workflows/compat-check-stable-stable.yml:50]:
```suggestion
    secrets: inherit

```

File: .github/workflows/compat-check-stable-stable.yml
Comment (maliasadi) [inline .github/workflows/compat-check-stable-stable.yml:1]: Could you also check stable/stable?

**Review by maliasadi — CHANGES_REQUESTED** [review]

File: .github/workflows/compat-check-stable-stable.yml
Comment (AmintorDusko) [reply .github/workflows/compat-check-stable-stable.yml:1]: yes

**Review by maliasadi — APPROVED** [review]
Thanks @AmintorDusko!

**Pull request merged**